server/src/main/java/org/opensearch/action/admin/cluster/node/reload/TransportNodesReloadSecureSettingsAction.java

```diff
@@ -216,11 +216,8 @@ private boolean isNodeTransportTLSEnabled() {
     }
 
     private boolean isNodeLocal(NodesReloadSecureSettingsRequest request) {
-        if (null == request.concreteNodes()) {
-            resolveRequest(request, clusterService.state());
-            assert request.concreteNodes() != null;
-        }
-        final DiscoveryNode[] nodes = request.concreteNodes();
+        final DiscoveryNode[] nodes = resolveRequest(request, clusterService.state());
```

Comment:
If the `clusterService.state()` changes b/w the co-ordinator making the request and datanode processing the request - wouldn't this impact ?

Reply:
In this specific case it shouldn't impact. From the clusterService.state() we are only identifying if the node on which the setting has to be applied is the local node. If the node is part of the cluster then the clusterService.state() will always contain the datanode on which the request needs to be executed and if the node is not part of the cluster then we don't want to process this transport request anyway.,

Reply:
If the value of node_id is set as `_local` wouldn't it matter if the the resolution happens on co-ordinator vs datanode ? Likewise for the value `_master` (or) `_cluster_manager`. Could we add testcases to validate the same ?